Spectrum_Peak_Visualization.py

```diff
@@ -0,0 +1,151 @@
+
+        end_region = np.flatnonzero(energy_axis > energy + Range)[0]
+        y = spectrum.data[start_region:end_region]
+        indexes = peakutils.indexes(y, thres=0.5, min_dist=4)
+        tallest_peak = []
+        if indexes.size == 0:
+            print('Peak Not Found')
+            skip += 1
+        else:
+            for i in range(indexes.size):
+                spot = spectrum.data[indexes[i]+start_region]
+                tallest_peak.append(spot)
+            indexes = indexes[np.argmax(tallest_peak)]
+            peak_channel.append(int(indexes+start_region))
+            found_energy.append(energy)
+            difference = abs((energy -
+                              float(energy_axis[int(indexes+start_region)])))
+            if difference > 0.5*fwhm:
+                fix += 1
+
+    """
+    Below is code directly from gamma analysis; have to modify the code to work
+    with code within the function.Loop through this code to obtain peak and 
+    background ROIs for each peak energy. Afterwards, will highlight each peak 
+    ROI as red and background ROI as green.
+    """
+    for energy in found_energy:
+        # Rough estimate of FWHM.
+        fwhm = 0.05*energy**0.5
+        
+        # Peak
+        start_peak = np.flatnonzero(energy_axis > energy - fwhm)[0]
+        end_peak = np.flatnonzero(energy_axis > energy + fwhm)[0]
+        plt.fill_between(energy_axis,counts,
+                         where= (energy_axis >= energy_axis[start_peak]) &
+                         (energy_axis <= energy_axis[end_peak]),facecolor='r')
+        
+        # Left Side of Peak
+        left_peak = energy - 2*fwhm
+        left_start = np.flatnonzero(energy_axis > left_peak - fwhm)[0]
+        left_end = np.flatnonzero(energy_axis > left_peak + fwhm)[0]
+        plt.fill_between(energy_axis,counts,
+                         where= (energy_axis >= energy_axis[left_start]) &
+                         (energy_axis <= energy_axis[left_end]),facecolor='g')
+        
+        # Right Side of Peak
+        right_peak = energy + 2*fwhm
+        right_start = np.flatnonzero(energy_axis > right_peak - fwhm)[0]
+        right_end = np.flatnonzero(energy_axis > right_peak + fwhm)[0]
+        plt.fill_between(energy_axis,counts,
+                         where= (energy_axis >= energy_axis[right_start]) &
+                         (energy_axis <= energy_axis[right_end]),facecolor='g')
+        
+    #Converting the y-axis to semilog.    
+    plt.yscale('log')
+    
+    """
+    If energy_range is set to a tuple of two values, then below code will graph
+    the spectrum with the highlighted peaks and background within the 
+    energy_range values for the energies (x-values).
+    """    
+    if energy_range != 'None':
+        plt.xlim([energy_range[0], energy_range[1]])
+
+"""
+#Test 1 for plot_spectrum
+measurement = SPEFile.SPEFile('UCB006_Bananas.Spe')
+measurement.read()
+plot_spectrum(measurement, 'UCB006_Bananas')
+"""
```

Comment:
Test code is good! Thank you! But please put it in functions like:
```
def test_1():
```
or even better, `def test_basic_plot():`, then `def test_energy_range_plot()`, or so.